#VRML_SIM R2023a utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/solids/protos/SolidBox.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.2427362815510867 -0.6958579465834746 0.6759147992131714 5.330329947562911
  position -1.6360570241063612 1.437936065412819 1.922178086258968
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  translation 0.05 0.05 0
  floorTileSize 0.2 0.2
}
Robot {
  children [
    DistanceSensor {
      translation 0.05 0 0.02
      name "DS_N"
      lookupTable [
        0 0 0
        0.3 0.3 0
      ]
    }
    DistanceSensor {
      translation 0 -0.05 0.02
      rotation 0 0 1 -1.5707950010791611
      name "DS_E"
      lookupTable [
        0 0 0
        0.3 0.3 0
      ]
    }
    DistanceSensor {
      translation -0.05 0 0.02
      rotation 0 0 1 3.141592653589793
      name "DS_S"
      lookupTable [
        0 0 0
        0.3 0.3 0
      ]
    }
    DistanceSensor {
      translation 0 0.05 0.02
      rotation 0 0 1 1.5707996938995747
      name "DS_W"
      lookupTable [
        0 0 0
        0.3 0.3 0
      ]
    }
    LED {
      translation 0.03 0 0.05
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "LED_N"
    }
    LED {
      translation 0 -0.03 0.05
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "LED_E"
    }
    LED {
      translation -0.03 0 0.05
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "LED_S"
    }
    LED {
      translation 0 0.03 0.05
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "LED_W"
    }
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
    Solid {
      boundingObject Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "bot1"
  controller "bot-controller"
  supervisor TRUE
}
Robot {
  translation 0.1 0 0
  children [
    DistanceSensor {
      translation 0.05 0 0.02
      name "DS_N"
      lookupTable [
        0 0 0
        0.3 0.3 0
      ]
    }
    DistanceSensor {
      translation 0 -0.05 0.02
      rotation 0 0 1 -1.5707950010791611
      name "DS_E"
      lookupTable [
        0 0 0
        0.3 0.3 0
      ]
    }
    DistanceSensor {
      translation -0.05 0 0.02
      rotation 0 0 1 3.141592653589793
      name "DS_S"
      lookupTable [
        0 0 0
        0.3 0.3 0
      ]
    }
    DistanceSensor {
      translation 0 0.05 0.02
      rotation 0 0 1 1.5707996938995747
      name "DS_W"
      lookupTable [
        0 0 0
        0.3 0.3 0
      ]
    }
    LED {
      translation 0.03 0 0.05
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "LED_N"
    }
    LED {
      translation 0 -0.03 0.05
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "LED_E"
    }
    LED {
      translation -0.03 0 0.05
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "LED_S"
    }
    LED {
      translation 0 0.03 0.05
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "LED_W"
    }
    Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
    Solid {
      boundingObject Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "bot2"
  controller "bot-controller"
  supervisor TRUE
}
Robot {
  translation 0 0.1 0
  children [
    DistanceSensor {
      translation 0.05 0 0.02
      name "DS_N"
      lookupTable [
        0 0 0
        0.3 0.3 0
      ]
    }
    DistanceSensor {
      translation 0 -0.05 0.02
      rotation 0 0 1 -1.5707950010791611
      name "DS_E"
      lookupTable [
        0 0 0
        0.3 0.3 0
      ]
    }
    DistanceSensor {
      translation -0.05 0 0.02
      rotation 0 0 1 3.141592653589793
      name "DS_S"
      lookupTable [
        0 0 0
        0.3 0.3 0
      ]
    }
    DistanceSensor {
      translation 0 0.05 0.02
      rotation 0 0 1 1.5707996938995747
      name "DS_W"
      lookupTable [
        0 0 0
        0.3 0.3 0
      ]
    }
    LED {
      translation 0.03 0 0.05
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "LED_N"
    }
    LED {
      translation 0 -0.03 0.05
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "LED_E"
    }
    LED {
      translation -0.03 0 0.05
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "LED_S"
    }
    LED {
      translation 0 0.03 0.05
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Capsule {
            height 0.01
            radius 0.01
          }
        }
      ]
      name "LED_W"
    }
    Shape {
      appearance PBRAppearance {
        baseColor 0 0 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
    Solid {
      boundingObject Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  name "bot3"
  controller "bot-controller"
  supervisor TRUE
}
SolidBox {
  translation -0.3 0.3 0.05
  size 0.1 0.3 0.1
}
SolidBox {
  translation -0.1 0.3 0.05
  name "box(1)"
  size 0.1 0.3 0.1
}
SolidBox {
  translation 0.2 0.3 0.05
  name "box(2)"
  size 0.1 0.3 0.1
}
SolidBox {
  translation 0.4 0.3 0.05
  name "box(3)"
  size 0.1 0.3 0.1
}
SolidBox {
  translation -0.3 -0.2 0.05
  name "box(4)"
  size 0.1 0.3 0.1
}
SolidBox {
  translation -0.1 -0.2 0.05
  name "box(5)"
  size 0.1 0.3 0.1
}
SolidBox {
  translation 0.2 -0.2 0.05
  name "box(6)"
  size 0.1 0.3 0.1
}
SolidBox {
  translation 0.4 -0.2 0.05
  name "box(7)"
  size 0.1 0.3 0.1
}
SolidBox {
  translation 1 0 0
  name "North"
  size 0.05 0.05 0.05
  appearance PBRAppearance {
    baseColor 0 0 0
    baseColorMap ImageTexture {
    }
    roughness 0.5
    metalness 0
  }
}
